src/observer.ts

```diff
@@ -324,14 +328,43 @@ export class DataLayerObserver {
     return handler;
   }
 
+  /**
+   * Schedules rule registration on a specified delay. The delay will be contingent on the number of attempts and the wait time.
+   * Each subsequent attempt at registration will use exponential backoff to lengthen the delay time between individual attempts.
+   * @param rule DataLayerRule to be registered
+   * @param attempt Numeric attempt counter, which starts at `1`
+   * @param wait Time in milliseconds to use as backoff multiplier
+   */
+  private registerRuleWithDelay(rule: DataLayerRule, attempt = 1, wait = 250) {
```

Comment:
Patrick, refactored out the registration logic.  Feels like this makes things a bit cleaner.

Reply:
Thank you. I've iterated on this. WDYT? https://github.com/fullstorydev/fullstory-data-layer-observer/pull/146